src/languages/html.coffee

```diff
@@ -0,0 +1,684 @@
+        bounds.end = {
+          line: lastLine
+          column: @lines[lastLine].length
+        }
+      else if node.__indentLocation.start is node.__indentLocation.end and node.__location.endTag
+        bounds.end = @positions[node.__location.endTag.start]
+
+    return bounds
+
+  trimText: (node) ->
+    text = node.value
+    location = node.__location
+    text = text.split '\n'
+    i = 0
+    while text[i].trim().length is 0
+      location.start += text[i].length + 1
+      i++
+    j = text.length - 1
+    while text[j].trim().length is 0
+      j--
+    text = text[i..j].join '\n'
+    location.end = location.start + text.length
+    if i isnt 0
+      leftTrimText = text.trimLeft()
+      location.start += text.length - leftTrimText.length
+      text = leftTrimText
+    node.value = text
+    ###
+    rigthtTrimText = text.trimRight()
+    leftTrimText = rigthtTrimText.trimLeft()
+    node.value = leftTrimText
+    location.start += rigthtTrimText.length - leftTrimText.length
+    location.end = location.start + leftTrimText.length
+    console.log node.value, JSON.stringify location
+    ###
+
+  getSocketLevel: (node) -> helper.ANY_DROP
+
+  htmlBlock: (node, depth, bounds) ->
+    #console.log "Adding Block: ", JSON.stringify(bounds ? @getBounds node)
+    @addBlock
+      bounds: bounds ? @getBounds node
+      depth: depth
+      precedence: @getPrecedence node
+      color: @getColor node
+      classes: @getClasses node
+      socketLevel: @getSocketLevel node
+
+  htmlSocket: (node, depth, precedence, bounds, classes) ->
+    #console.log "Adding Socket: ", JSON.stringify(bounds ? @getBounds node)
+    @addSocket
+      bounds: bounds ? @getBounds node
+      depth: depth
+      precedence: precedence
+      classes: classes ? @getClasses node
+      acccepts: @getAcceptsRule node
+
+  getIndentPrefix: (bounds, indentDepth, depth) ->
+    #console.log JSON.stringify(bounds), indentDepth, depth
+    if bounds.end.line - bounds.start.line < 1
+      return DEFAULT_INDENT_DEPTH
+    else
+      tmp = 1
+      while @lines[bounds.start.line + tmp].trim().length is 0
+        tmp++
+      line = @lines[bounds.start.line + tmp]
+      lineIndent = line[indentDepth...(line.length - line.trimLeft().length)]
+      if lineIndent.length >= DEFAULT_INDENT_DEPTH.length
+        return lineIndent
+      else
+        return DEFAULT_INDENT_DEPTH
+
+  handleText: (node, depth) ->
+    text = node.value.split '\n'
+    loc = {start: node.__location.start}
+    for line in text
+      loc.end = loc.start + line.length
+      @htmlSocket node, depth, null, @genBounds loc
+      loc.start = loc.end + 1
+
+  markRoot: ->
+    @positions = []
+    line = 0
+    column = 0
+    for val, i in @text
+      @positions[i] = {'line': line, 'column': column}
+      column++
+      if val is '\n'
+        line++
+        column = 0
+    @positions[@text.length] = {'line': line, 'column': column}
+    window.positions = @positions
```

Comment:
It seems like this was probably used for debugging and you just forgot to remove it?